h3/src/lib.rs

```diff
@@ -6,7 +6,7 @@ pub mod server;
 
 #[allow(dead_code)]
 mod connection;
-mod frame;
+pub mod frame;
```

Comment:
Is this change needed in the current PR ?

Reply:
probably not necessary to update to quin 0.8, but I needed it because I passed the connection as a method argument and had to type the fully qualified path out in my downstream project. since it wasn't public it wasn't letting me do that.

Reply:
to be clear, I have this line written out and used as arguments throughout my program. 

`type RequestStreamTyped = RequestStream<FrameStream<BidiStream<Bytes>>>;`

Reply:
The API is not very ergonomic :thinking:

Reply:
since it’s not integrated with hyper or warp yet I wanted to pass the stream to my own hand rolled request matcher function. if I did it inline, not requiring the full definition, then I would have been 5 of 6 indents in. I never needed to interact with the frame itself, so maybe this meta type could just be exported in h3.

Reply:
If we’re talking about ergonomics though, the body being hard coded as () and then using a separate method for the real body is weird. It seems like you should be able to specify one or the other. It took me a few tries to figure out I was supposed to ignore the () body and use the other method.

Reply:
Regarding the body, you are right. But if you want to stream it, using the `Request<>` associated type makes it kind of hard.
For now there is no docs because the API is subject to change. We need to find an API that is suitable for Hyper (and Actix), as well as ergonomic. Even if this crate is low-level.

I'll open an issue today to address the `FrameStream<>` ergonomics problem separately.

Thanks a lot for your feedback, it is very valuable. BTW, is your project available somewhere, so I can have a look ?

Reply:
unfortunately I can’t share the downstream project, but a relatively standard http3 server with a request router method and and auth filter in front of it would get you close in terms of the h3 interaction. Since you’re going to address the issue separately would you like me to remove the pub from here?